plaso/parsers/fseventsd.py

```diff
@@ -172,15 +181,26 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
       timestamp_description = definitions.TIME_DESCRIPTION_NOT_A_TIME
     event = time_events.DateTimeValuesEvent(date_time, timestamp_description)
 
-    while file_object.get_offset() < file_object.get_size():
-      if file_object.get_offset() >= current_page_end:
-        page_header = self._ParseDLSPageHeader(file_object)
+    while file_offset < file_size:
+      if file_offset >= current_page_end:
+        page_header, header_size = self._ParseDLSPageHeader(
+            file_object, file_offset)
         current_page_end += page_header.page_size
+        file_offset += header_size
         continue
       if page_header.signature == self._DLS_V1_SIGNATURE:
-        record = self._DLS_RECORD_V1.parse_stream(file_object)
+        record_map = self._GetDataTypeMap('dls_record_v1')
       else:
-        record = self._DLS_RECORD_V2.parse_stream(file_object)
+        record_map = self._GetDataTypeMap('dls_record_v2')
+
+      try:
+        record, record_length = self._ReadStructureFromFileObject(
+            file_object, file_offset, record_map)
+        file_offset += record_length
+      except (ValueError, errors.ParseError) as exception:
+        raise errors.UnableToParseFile(
```

Comment:
Create an extraction error here, seeing the some records were successfully parsed and the file does not need to be tried by other parsers